homebrew-macos-26-minimal: libhomfly selects `std=gnu23` and fails

```
  [libhomfly-1.03]   [spkg-install] Configuring libhomfly-1.03
  [libhomfly-1.03]   [spkg-install] configure: WARNING: unrecognized options: --disable-maintainer-mode
  [libhomfly-1.03]   [spkg-install] checking for a BSD-compatible install... /usr/bin/install -c
  [libhomfly-1.03]   [spkg-install] checking whether sleep supports fractional seconds... yes
  [libhomfly-1.03]   [spkg-install] checking filesystem timestamp resolution... 2
  [libhomfly-1.03]   [spkg-install] checking whether build environment is sane... yes
  [libhomfly-1.03]   [spkg-install] checking for a race-free mkdir -p... mkdir -p
  [libhomfly-1.03]   [spkg-install] checking for gawk... no
  [libhomfly-1.03]   [spkg-install] checking for mawk... no
  [libhomfly-1.03]   [spkg-install] checking for nawk... no
  [libhomfly-1.03]   [spkg-install] checking for awk... awk
  [libhomfly-1.03]   [spkg-install] checking whether make sets $(MAKE)... yes
  [libhomfly-1.03]   [spkg-install] checking whether make supports nested variables... yes
  [libhomfly-1.03]   [spkg-install] checking xargs -n works... yes
  [libhomfly-1.03]   [spkg-install] checking whether UID '501' is supported by ustar format... yes
  [libhomfly-1.03]   [spkg-install] checking whether GID '20' is supported by ustar format... yes
  [libhomfly-1.03]   [spkg-install] checking how to create a ustar tar archive... gnutar
  [libhomfly-1.03]   [spkg-install] checking whether make supports the include directive... yes (GNU style)
  [libhomfly-1.03]   [spkg-install] checking for gcc... gcc -std=gnu23
  [libhomfly-1.03]   [spkg-install] checking whether the C compiler works... yes
  [libhomfly-1.03]   [spkg-install] checking for C compiler default output file name... a.out
  [libhomfly-1.03]   [spkg-install] checking for suffix of executables... 
  [libhomfly-1.03]   [spkg-install] checking whether we are cross compiling... no
  [libhomfly-1.03]   [spkg-install] checking for suffix of object files... o
  [libhomfly-1.03]   [spkg-install] checking whether the compiler supports GNU C... yes
  [libhomfly-1.03]   [spkg-install] checking whether gcc -std=gnu23 accepts -g... yes
  [libhomfly-1.03]   [spkg-install] checking for gcc -std=gnu23 option to enable C11 features... none needed
  [libhomfly-1.03]   [spkg-install] checking whether gcc -std=gnu23 understands -c and -o together... yes
  [libhomfly-1.03]   [spkg-install] checking dependency style of gcc -std=gnu23... none
  [libhomfly-1.03]   [spkg-install] checking the archiver (ar) interface... ar
  [libhomfly-1.03]   [spkg-install] checking for gcc... (cached) gcc -std=gnu23
  [libhomfly-1.03]   [spkg-install] checking whether the compiler supports GNU C... (cached) yes
  [libhomfly-1.03]   [spkg-install] checking whether gcc -std=gnu23 accepts -g... (cached) yes
  [libhomfly-1.03]   [spkg-install] checking for gcc -std=gnu23 option to enable C11 features... (cached) none needed
  [libhomfly-1.03]   [spkg-install] checking whether gcc -std=gnu23 understands -c and -o together... (cached) yes
  [libhomfly-1.03]   [spkg-install] checking dependency style of gcc -std=gnu23... (cached) none
  [libhomfly-1.03]   [spkg-install] checking build system type... aarch64-apple-darwin25.3.0
  [libhomfly-1.03]   [spkg-install] checking host system type... aarch64-apple-darwin25.3.0
  [libhomfly-1.03]   [spkg-install] checking how to print strings... printf
  [libhomfly-1.03]   [spkg-install] checking for a sed that does not truncate output... /usr/bin/sed
  [libhomfly-1.03]   [spkg-install] checking for grep that handles long lines and -e... /usr/bin/grep
  [libhomfly-1.03]   [spkg-install] checking for egrep... /usr/bin/grep -E
  [libhomfly-1.03]   [spkg-install] checking for fgrep... /usr/bin/grep -F
  [libhomfly-1.03]   [spkg-install] checking for ld used by gcc -std=gnu23... /Applications/Xcode_26.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
  [libhomfly-1.03]   [spkg-install] checking if the linker (/Applications/Xcode_26.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
  [libhomfly-1.03]   [spkg-install] checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
  [libhomfly-1.03]   [spkg-install] checking the name lister (/usr/bin/nm -B) interface... BSD nm
  [libhomfly-1.03]   [spkg-install] checking whether ln -s works... yes
  [libhomfly-1.03]   [spkg-install] checking the maximum length of command line arguments... 786432
  [libhomfly-1.03]   [spkg-install] checking how to convert aarch64-apple-darwin25.3.0 file names to aarch64-apple-darwin25.3.0 format... func_convert_file_noop
  [libhomfly-1.03]   [spkg-install] checking how to convert aarch64-apple-darwin25.3.0 file names to toolchain format... func_convert_file_noop
  [libhomfly-1.03]   [spkg-install] checking for /Applications/Xcode_26.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
  [libhomfly-1.03]   [spkg-install] checking for file... file
  [libhomfly-1.03]   [spkg-install] checking for objdump... objdump
  [libhomfly-1.03]   [spkg-install] checking how to recognize dependent libraries... pass_all
  [libhomfly-1.03]   [spkg-install] checking for dlltool... no
  [libhomfly-1.03]   [spkg-install] checking how to associate runtime and link libraries... printf %s\n
  [libhomfly-1.03]   [spkg-install] checking for ranlib... ranlib
  [libhomfly-1.03]   [spkg-install] checking for archiver @FILE support... no
  [libhomfly-1.03]   [spkg-install] checking for strip... strip
  [libhomfly-1.03]   [spkg-install] checking command to parse /usr/bin/nm -B output from gcc -std=gnu23 object... ok
  [libhomfly-1.03]   [spkg-install] checking for sysroot... no
  [libhomfly-1.03]   [spkg-install] checking for a working dd... /bin/dd
  [libhomfly-1.03]   [spkg-install] checking how to truncate binary pipes... /bin/dd bs=4096 count=1
  [libhomfly-1.03]   [spkg-install] checking for mt... no
  [libhomfly-1.03]   [spkg-install] checking if : is a manifest tool... no
  [libhomfly-1.03]   [spkg-install] checking for dsymutil... dsymutil
  [libhomfly-1.03]   [spkg-install] checking for nmedit... nmedit
  [libhomfly-1.03]   [spkg-install] checking for lipo... lipo
  [libhomfly-1.03]   [spkg-install] checking for otool... otool
  [libhomfly-1.03]   [spkg-install] checking for otool64... no
  [libhomfly-1.03]   [spkg-install] checking for -single_module linker flag... ld: warning: -single_module is obsolete
  [libhomfly-1.03]   [spkg-install] no
  [libhomfly-1.03]   [spkg-install] checking for -no_fixup_chains linker flag... yes
  [libhomfly-1.03]   [spkg-install] checking for -exported_symbols_list linker flag... yes
  [libhomfly-1.03]   [spkg-install] checking for -force_load linker flag... yes
  [libhomfly-1.03]   [spkg-install] checking for stdio.h... yes
  [libhomfly-1.03]   [spkg-install] checking for stdlib.h... yes
  [libhomfly-1.03]   [spkg-install] checking for string.h... yes
  [libhomfly-1.03]   [spkg-install] checking for inttypes.h... yes
  [libhomfly-1.03]   [spkg-install] checking for stdint.h... yes
  [libhomfly-1.03]   [spkg-install] checking for strings.h... yes
  [libhomfly-1.03]   [spkg-install] checking for sys/stat.h... yes
  [libhomfly-1.03]   [spkg-install] checking for sys/types.h... yes
  [libhomfly-1.03]   [spkg-install] checking for unistd.h... yes
  [libhomfly-1.03]   [spkg-install] checking for dlfcn.h... yes
  [libhomfly-1.03]   [spkg-install] checking for objdir... .libs
  [libhomfly-1.03]   [spkg-install] checking if gcc -std=gnu23 supports -fno-rtti -fno-exceptions... yes
  [libhomfly-1.03]   [spkg-install] checking for gcc -std=gnu23 option to produce PIC... -fno-common -DPIC
  [libhomfly-1.03]   [spkg-install] checking if gcc -std=gnu23 PIC flag -fno-common -DPIC works... yes
  [libhomfly-1.03]   [spkg-install] checking if gcc -std=gnu23 static flag -static works... no
  [libhomfly-1.03]   [spkg-install] checking if gcc -std=gnu23 supports -c -o file.o... yes
  [libhomfly-1.03]   [spkg-install] checking if gcc -std=gnu23 supports -c -o file.o... (cached) yes
  [libhomfly-1.03]   [spkg-install] checking whether the gcc -std=gnu23 linker (/Applications/Xcode_26.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
  [libhomfly-1.03]   [spkg-install] checking dynamic linker characteristics... darwin25.3.0 dyld
  [libhomfly-1.03]   [spkg-install] checking how to hardcode library paths into programs... immediate
  [libhomfly-1.03]   [spkg-install] checking whether stripping libraries is possible... yes
  [libhomfly-1.03]   [spkg-install] checking if libtool supports shared libraries... yes
  [libhomfly-1.03]   [spkg-install] checking whether to build shared libraries... yes
  [libhomfly-1.03]   [spkg-install] checking whether to build static libraries... no
  [libhomfly-1.03]   [spkg-install] checking that generated files are newer than configure... done
  [libhomfly-1.03]   [spkg-install] configure: creating ./config.status
  [libhomfly-1.03]   [spkg-install] config.status: creating Makefile
  [libhomfly-1.03]   [spkg-install] config.status: creating lib/Makefile
  [libhomfly-1.03]   [spkg-install] config.status: creating test/Makefile
  [libhomfly-1.03]   [spkg-install] config.status: creating libhomfly.pc
  [libhomfly-1.03]   [spkg-install] config.status: executing depfiles commands
  [libhomfly-1.03]   [spkg-install] config.status: executing libtool commands
  [libhomfly-1.03]   [spkg-install] configure: WARNING: unrecognized options: --disable-maintainer-mode
  [libhomfly-1.03]   [spkg-install] Building libhomfly-1.03
  [libhomfly-1.03]   [spkg-install] Making all in lib
  [libhomfly-1.03]   [spkg-install] /bin/bash ../libtool  --tag=CC   --mode=compile gcc -std=gnu23 -DPACKAGE_NAME=\"libhomfly\" -DPACKAGE_TARNAME=\"libhomfly\" -DPACKAGE_VERSION=\"1.03\" -DPACKAGE_STRING=\"libhomfly\ 1.03\" -DPACKAGE_BUGREPORT=\"mmarco@unizar.es\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libhomfly\" -DVERSION=\"1.03\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I.     -g -O2 -c -o bound.lo bound.c
  [libhomfly-1.03]   [spkg-install] libtool: compile:  gcc -std=gnu23 -DPACKAGE_NAME=\"libhomfly\" -DPACKAGE_TARNAME=\"libhomfly\" -DPACKAGE_VERSION=\"1.03\" "-DPACKAGE_STRING=\"libhomfly 1.03\"" -DPACKAGE_BUGREPORT=\"mmarco@unizar.es\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libhomfly\" -DVERSION=\"1.03\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I. -g -O2 -c bound.c  -fno-common -DPIC -o .libs/bound.o
  [libhomfly-1.03]   [spkg-install] bound.c:195:17: error: unknown type name 'list'
  [libhomfly-1.03]   [spkg-install]   195 | void b_one_pair(list, list2, one, two)
  [libhomfly-1.03]   [spkg-install]       |                 ^
  [libhomfly-1.03]   [spkg-install] bound.c:195:23: error: unknown type name 'list2'
  [libhomfly-1.03]   [spkg-install]   195 | void b_one_pair(list, list2, one, two)
  [libhomfly-1.03]   [spkg-install]       |                       ^
  [libhomfly-1.03]   [spkg-install] bound.c:195:30: error: unknown type name 'one'
  [libhomfly-1.03]   [spkg-install]   195 | void b_one_pair(list, list2, one, two)
  [libhomfly-1.03]   [spkg-install]       |                              ^
  [libhomfly-1.03]   [spkg-install] bound.c:195:35: error: unknown type name 'two'
  [libhomfly-1.03]   [spkg-install]   195 | void b_one_pair(list, list2, one, two)
  [libhomfly-1.03]   [spkg-install]       |                                   ^
  [libhomfly-1.03]   [spkg-install] bound.c:195:39: error: expected ';' after top level declarator
  [libhomfly-1.03]   [spkg-install]   195 | void b_one_pair(list, list2, one, two)
  [libhomfly-1.03]   [spkg-install]       |                                       ^
  [libhomfly-1.03]   [spkg-install]       |                                       ;
  [libhomfly-1.03]   [spkg-install] bound.c:196:8: error: redefinition of 'list' with a different type: 'word *' (aka 'int *') vs 'word[26]' (aka 'int[26]')
  [libhomfly-1.03]   [spkg-install]   196 | word  *list;    /* list of original inputs/outputs, modified by this routine */
  [libhomfly-1.03]   [spkg-install]       |        ^
  [libhomfly-1.03]   [spkg-install] bound.c:24:6: note: previous definition is here
  [libhomfly-1.03]   [spkg-install]    24 | word list[BIGWEAVE];              /* description of first new weave */
  [libhomfly-1.03]   [spkg-install]       |      ^
  [libhomfly-1.03]   [spkg-install] bound.c:197:8: error: redefinition of 'list2' with a different type: 'word *' (aka 'int *') vs 'word[26]' (aka 'int[26]')
  [libhomfly-1.03]   [spkg-install]   197 | word  *list2;                      /* inputs/outputs of the second new weave */
  [libhomfly-1.03]   [spkg-install]       |        ^
  [libhomfly-1.03]   [spkg-install] bound.c:25:6: note: previous definition is here
  [libhomfly-1.03]   [spkg-install]    25 | word list2[BIGWEAVE];           /* description of second, if needed */
  [libhomfly-1.03]   [spkg-install]       |      ^
  [libhomfly-1.03]   [spkg-install] bound.c:200:1: error: expected identifier or '('
  [libhomfly-1.03]   [spkg-install]   200 | {
  [libhomfly-1.03]   [spkg-install]       | ^
  [libhomfly-1.03]   [spkg-install] 8 errors generated.
  [libhomfly-1.03]   [spkg-install] make[4]: *** [bound.lo] Error 1
```
@miguelmarco

--- a/build/pkgs/glpk/spkg-install.in
+++ b/build/pkgs/glpk/spkg-install.in
@@ -24,9 +24,10 @@ fi
 #          are valid options.  So we *have to* add Sage's include and
 #          library directories to `CPPFLAGS` (done here) and `LDFLAGS`
 #          (already the default):
-CPPFLAGS="-I$SAGE_LOCAL/include $CPPFLAGS"
+export CPPFLAGS="-I$SAGE_LOCAL/include $CPPFLAGS"
 
-export CFLAGS LDFLAGS CPPFLAGS
+# Compilation errors with std=gnu23, https://github.com/passagemath/passagemath/issues/2308
+export CFLAGS="-std=gnu17 $CFLAGS"
 
 sdh_configure --with-gmp --disable-static
 sdh_make

--- a/build/pkgs/libgd/spkg-install.in
+++ b/build/pkgs/libgd/spkg-install.in
@@ -2,6 +2,9 @@ cd src
 
 export CFLAGS="-g $CFLAGS"
 
+# Compilation errors in libgd 2.3.3 with std=gnu23, https://github.com/passagemath/passagemath/issues/2307
+export CFLAGS="-std=gnu17 $CFLAGS"
+
 cp "$SAGE_ROOT"/config/config.* config/
 
 if [ -n "$MSYSTEM" ]; then

--- a/build/pkgs/libhomfly/spkg-install.in
+++ b/build/pkgs/libhomfly/spkg-install.in
@@ -1,5 +1,9 @@
 cd src
 cp "$SAGE_ROOT"/config/config.* build-aux/
+
+# Compilation errors with std=gnu23, https://github.com/passagemath/passagemath/issues/2306
+export CFLAGS="-std=gnu17 $CFLAGS"
+
 sdh_configure $LIBHOMFLY_CONFIGURE
 sdh_make
 sdh_make_install -j1

